go/oasis-node/cmd/debug/txsource/workload/runtime.go

```diff
@@ -0,0 +1,230 @@
+		"request", req,
+	)
+
+	// Wait for a maximum of 30 seconds as invalid submissions may block
+	// forever.
+	submitCtx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	out, err := rtc.SubmitTx(submitCtx, rtx)
+	if err != nil {
+		cancel()
+		return nil, fmt.Errorf("failed to submit runtime transaction: %w", err)
+	}
+	cancel()
+
+	if err = cbor.Unmarshal(out, &rsp); err != nil {
+		return nil, fmt.Errorf("malformed tx output from runtime: %w", err)
+	}
+	if rsp.Error != nil {
+		return nil, fmt.Errorf("runtime tx failed: %s", *rsp.Error)
+	}
+
+	return &rsp, nil
+}
+
+func (r *runtime) doInsertRequest(ctx context.Context, rng *rand.Rand, rtc runtimeClient.RuntimeClient, existing bool) error {
+	key := r.generateVal(rng, existing)
+	value := r.generateVal(rng, false)
+
+	// Submit request.
+	rsp, err := r.submitRuntimeRquest(ctx, rtc, &runtimeTransaction.TxnCall{
+		Method: "insert",
+		Args: struct {
+			Key   string `json:"key"`
+			Value string `json:"value"`
+		}{
+			Key:   key,
+			Value: value,
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("submit insert request failed: %w", err)
+	}
+	r.logger.Debug("insert request success",
+		"response", rsp,
+	)
+
+	if err := r.validateResponse(key, rsp); err != nil {
+		return fmt.Errorf("invalid response: %w", err)
+	}
+
+	// Update local state.
+	r.state[key] = value
+
+	return nil
+}
+
+func (r *runtime) doGetRequest(ctx context.Context, rng *rand.Rand, rtc runtimeClient.RuntimeClient, existing bool) error {
+	key := r.generateVal(rng, existing)
+
+	// Submit request.
+	rsp, err := r.submitRuntimeRquest(ctx, rtc, &runtimeTransaction.TxnCall{
+		Method: "get",
+		Args:   key,
+	})
+	if err != nil {
+		return fmt.Errorf("submit get request failed: %w", err)
+	}
+	r.logger.Debug("get request success",
+		"response", rsp,
+	)
+
+	if err := r.validateResponse(key, rsp); err != nil {
+		return fmt.Errorf("invalid response: %w", err)
+	}
+
+	return nil
+}
+
+func (r *runtime) doRemoveRequest(ctx context.Context, rng *rand.Rand, rtc runtimeClient.RuntimeClient, existing bool) error {
+	key := r.generateVal(rng, existing)
+
+	// Submit request.
+	rsp, err := r.submitRuntimeRquest(ctx, rtc, &runtimeTransaction.TxnCall{
+		Method: "remove",
+		Args:   key,
+	})
+	if err != nil {
+		return fmt.Errorf("submit remove request failed: %w", err)
+	}
+	r.logger.Debug("remove request success",
+		"response", rsp,
+	)
+
+	if err := r.validateResponse(key, rsp); err != nil {
+		return fmt.Errorf("invalid response: %w", err)
+	}
+
+	// Update local state.
+	delete(r.state, key)
+
+	return nil
+}
+
+func (r *runtime) Run(gracefulExit context.Context, rng *rand.Rand, _ *grpc.ClientConn, _ consensus.ClientBackend, rtc runtimeClient.RuntimeClient, fundingAccount signature.Signer) error {
+	ctx := context.Background()
+
+	r.logger = logging.GetLogger("cmd/txsource/workload/runtime")
+	// Simple-keyvalue runtime.
+	_ = r.runtimeID.UnmarshalHex("8000000000000000000000000000000000000000000000000000000000000000")
```

Comment:
Could this be changed to be a CLI argument instead of being hardcoded here? And then passed from txsource E2E test.

Reply:
:+1: